plugins/kitpvp-paper/src/main/java/dev/drekamor/warp/database/DatabaseManager.java

```diff
@@ -0,0 +1,135 @@
+package dev.drekamor.warp.database;
+
+import dev.drekamor.warp.util.Warp;
+import net.civmc.kitpvp.KitPvpPlugin;
+import vg.civcraft.mc.civmodcore.dao.ManagedDatasource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatabaseManager {
+    private final KitPvpPlugin plugin;
+    private final ManagedDatasource dataSource;
+
+    public DatabaseManager (KitPvpPlugin plugin, ManagedDatasource dataSource) {
+        this.plugin = plugin;
+        this.dataSource = dataSource;
+        this.plugin.info("Initialised a database connection");
+
+
+        initialiseTables();
+        dataSource.updateDatabase();
+    }
+
+    public Connection getConnection() {
+        try {
+            return this.dataSource.getConnection();
+        } catch (SQLException e) {
+            plugin.getLogger().severe("Failed to obtain a data source");
+            plugin.getLogger().severe(Arrays.toString(e.getStackTrace()));
+        }
+        return null;
+    }
+
+    private void initialiseTables() {
+        dataSource.registerMigration(2, false, """
+            CREATE TABLE IF NOT EXISTS warps (
+                name VARCHAR(64) UNIQUE PRIMARY KEY NOT NULL,
```

Comment:
```suggestion
                name VARCHAR(64) PRIMARY KEY,
```
UNIQUE and NOT NULL are implied by PRIMARY KEY